xapian-core/cluster/kmeans.cc

```diff
@@ -72,7 +79,7 @@ void
 KMeans::initialise_points(const MSet &source)
 {
     LOGCALL_VOID(API, "KMeans::initialise_points", source);
-    TermListGroup tlg(source);
+    tlg.construct_tlg(source);
```

Comment:
So the reason I did it like this is because of storing the stopper as a pointer rather than an object. I wasn't able to pass the stopper object around because it is an `opt_intrusive_ptr` and passing it as a pointer of Stopper class raised an error. So it made sense to make it a member of TermListGroup because it would only be used there.

> This prolongs the lifetime of the TermListGroup and the (potentially large) unordered_map it contains.
It does make sense to thus add it in KMeans itself due to the map being potentially huge. But in that case, I'm confused about whether I should be storing the Stopper as an object or as a pointer. I saw the previous implementations in the queryparser where it has been stored as a pointer and followed the same. Is there a specific reason it was stored that way? Or would it not harm to have the Stopper object itself?